### PR TITLE
ByteCount using asizeof

### PR DESCRIPTION
**Pull request by Piruzzolo**
Implements ByteCount using the BSD-licensed asizeof module, which scans through the object tree and gives the total byte count of it. The results may differ between Python implementations and versions.

### CODE REVIEW

**Comment by Piruzzolo** [comment]
see #561 for memory issues found

**Comment by Piruzzolo** [comment]
@sn6uv There seems to be a problem with pypy build. Could you please take a look? Maybe pypy is not compatible with CPython in this case?

File: mathics/builtin/structure.py [hunk 1]
```diff
@@ -1276,5 +1281,10 @@ class ByteCount(Builtin):
 
     def apply(self, expression, evaluation):
         'ByteCount[expression_]'
-
-        return Integer(count_bytes(expression))
+        if bytecount_support == False:
```
Comment (sn6uv) [inline mathics/builtin/structure.py]: I thought it was just PyPy that didn't support bytecount?
In that case:
```
if platform.python_implementation == 'PyPy':
    bytecount_supporty = False
else:
    import ...
    bytecount_support = True
```
and then
```
if not bytecount_support:
    return evaluation.message('ByteCount', 'pypy')
```
would be simpler.

Comment (Piruzzolo) [inline mathics/builtin/structure.py]: As of january 2016, Jython [has](https://mail.python.org/pipermail/jython-checkins/2016-January/001669.html) getsizeof() support too. I think it's just PyPy.

Comment (Piruzzolo) [inline mathics/builtin/structure.py]: I think it's just PyPy. Jython too [supports](https://mail.python.org/pipermail/jython-checkins/2016-January/001669.html) sys.getsizeof().

**Comment by TiberiumPY** [comment]
PyPy will not support getsizeof in future - because this is hard to implement and there's no good way to get REAL memory usage (because of custom GC and JIT)

File: mathics/builtin/structure.py [hunk 2]
```diff
@@ -16,11 +16,12 @@
 import six
 import platform
 from six.moves import range
-try:
+
+if platform.python_implementation == 'PyPy':
```
Comment (sn6uv) [inline mathics/builtin/structure.py]: I think this needs to be a function call, .i.e.
```
>>>> platform.python_implementation()
'PyPy'
```

**Pull request merged**